src/CommonLib/Cache.cs

```diff
@@ -125,54 +122,64 @@ private static string GetPrefixKey(string key, string domain)
             return $"{key}|{domain}";
         }
 
-        public static void CreateNewCache()
+        public static Cache CreateNewCache()
         {
-            CacheInstance = new Cache();
+            return new Cache();
         }
 
-        private static string GetCacheStats()
+        public static void SetCacheInstance(Cache cache)
+        {
+            CacheInstance = cache;
+        }
+
+        public string GetCacheStats()
         {
             try
             {
                 return
-                    $"{CacheInstance._idToTypeCache.Count} ID to type mappings.\n {CacheInstance._valueToIDCache.Count} name to SID mappings.\n {CacheInstance._machineSidCache.Count} machine sid mappings.\n {CacheInstance._sidToDomainCache.Count} sid to domain mappings.\n {CacheInstance._globalCatalogCache.Count} global catalog mappings.";
+                    $"{_idToTypeCache.Count} ID to type mappings.\n {_valueToIDCache.Count} name to SID mappings.\n {_machineSidCache.Count} machine sid mappings.\n {_sidToDomainCache.Count} sid to domain mappings.\n {_globalCatalogCache.Count} global catalog mappings.";
             }
             catch
             {
                 return "";
             }
         }
 
-        public static void LoadExistingCache(string filePath)
-        {
-            if (!File.Exists(filePath))
-            {
-                CacheInstance = new Cache();
-                Logging.Debug("Cache file not found, empty cache created.");
-                return;
-            }
-
-            try
-            {
-                Logging.Debug($"Loading cache from {filePath}");
-                var bytes = File.ReadAllBytes(filePath);
-                var json = new UTF8Encoding(true).GetString(bytes);
-                CacheInstance = JsonConvert.DeserializeObject<Cache>(json, new JsonSerializerSettings
-                {
-                    DefaultValueHandling = DefaultValueHandling.Populate
-                });
-            }
-            catch (Exception e)
-            {
-                Logging.Debug($"Exception loading cache: {e}. Creating empty cache.");
-                CacheInstance = new Cache();
-            }
-
-            CreateMissingDictionaries();
-
-            Logging.Debug(
-                $"Cache file loaded!\n {GetCacheStats()}");
-        }
+        public static Cache GetCacheInstance()
+        {
+            return CacheInstance;
+        }
+
+        // public static void LoadExistingCache(string filePath)
```

Comment:
remove commented out code or include a comment

Reply:
Luckily, no one uses this yet. I will be bumping the version to 1.1.0 though